### PR TITLE
Fix Portworx CSI storage capabilities

### PR DESCRIPTION
**Pull request by arnongilboa**
**What this PR does / why we need it**:
Add preferred `ReadWriteMany, Filesystem` capability to the `pxd.portworx.com` provisioner.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
Add preferred ReadWriteMany, Filesystem capability to the pxd.portworx.com provisioner
```

### CODE REVIEW

File: pkg/storagecapabilities/storagecapabilities.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ var CapabilitiesByProvisionerKey = map[string][]StorageCapabilities{
 	"pxd.openstorage.org/shared": createOpenStorageSharedVolumeCapabilities(),
 	"pxd.openstorage.org":        createRWOBlockAndFilesystemCapabilities(),
 	"pxd.portworx.com/shared":    createOpenStorageSharedVolumeCapabilities(),
-	"pxd.portworx.com":           createRWOBlockAndFilesystemCapabilities(),
+	"pxd.portworx.com":           {{rwx, file}, {rwo, block}, {rwo, file}},
```
[the remark this answers is not in the document]
Comment (orenc1) [reply pkg/storagecapabilities/storagecapabilities.go:92]: no need for `shared: true` in the storage class parameters for it to be ReadWriteMany, as far as I saw.
I used a PVC of Filesystem ReadWriteMany with `px-csi-db` SC.

File: pkg/storagecapabilities/storagecapabilities.go [hunk 2]
```diff
@@ -89,7 +89,7 @@ var CapabilitiesByProvisionerKey = map[string][]StorageCapabilities{
 	"pxd.openstorage.org/shared": createOpenStorageSharedVolumeCapabilities(),
 	"pxd.openstorage.org":        createRWOBlockAndFilesystemCapabilities(),
 	"pxd.portworx.com/shared":    createOpenStorageSharedVolumeCapabilities(),
```
Comment (aglitke) [inline pkg/storagecapabilities/storagecapabilities.go:91]: Noticed that we are saying RWX/Block is supported but https://github.com/libopenstorage/openstorage/blob/master/csi/controller.go#L408-L412 begs to differ?

Comment (aglitke) [reply pkg/storagecapabilities/storagecapabilities.go:91]: Seems we may need to revamp and correct the portworx caps and backport to previous releases too.

Comment (arnongilboa) [reply pkg/storagecapabilities/storagecapabilities.go:91]: Ok, so the code impliciltly says RWX/File,RWO/Block,RWO/File are supported in the shared provisioners?

File: pkg/storagecapabilities/storagecapabilities.go [hunk 3]
```diff
@@ -87,9 +87,9 @@ var CapabilitiesByProvisionerKey = map[string][]StorageCapabilities{
 	"kubernetes.io/portworx-volume":        {{rwo, file}},
 	// Portworx CSI
 	"pxd.openstorage.org/shared": createOpenStorageSharedVolumeCapabilities(),
-	"pxd.openstorage.org":        createRWOBlockAndFilesystemCapabilities(),
+	"pxd.openstorage.org":        createOpenStorageSharedVolumeCapabilities(),
 	"pxd.portworx.com/shared":    createOpenStorageSharedVolumeCapabilities(),
-	"pxd.portworx.com":           createRWOBlockAndFilesystemCapabilities(),
+	"pxd.portworx.com":           createOpenStorageSharedVolumeCapabilities(),
```
Comment (jpeimer) [inline pkg/storagecapabilities/storagecapabilities.go:92]: Created DVs on `pxd.portworx.com` - all these Succeeded: `{RWX, FS}, {RWO, Block}, {RWO, FS}`

**Review by jpeimer — APPROVED** [review]

**Comment by akalenyu** [comment]
/cc @pureneelesh @luayalem

Hey! We are trying to provide some defaults for Portworx storage classes
Does this look right?
https://github.com/kubevirt/containerized-data-importer/blob/e5ba88e189172a1bb030a28c98a4eacee3ddf29c/pkg/storagecapabilities/storagecapabilities.go#L298-L304

Also IIUC "shared" is deprecated, and our code still filters based on that
https://github.com/kubevirt/containerized-data-importer/blob/e5ba88e189172a1bb030a28c98a4eacee3ddf29c/pkg/storagecapabilities/storagecapabilities.go#L234-L240

**Comment by akalenyu** [comment]
Following offline discussion, we will pursue pinpointed portworx variation filtering outside of this PR.
For now, this is fixing an error (rwx, block) and will benefit users.

/approve

**Pull request merged**